plugins/utils/todo-comments.nix

```diff
@@ -1,290 +1,429 @@
 {
   lib,
-  helpers,
   config,
   pkgs,
   ...
 }:
 with lib;
 let
-  cfg = config.plugins.todo-comments;
-
-  commands = {
+  inherit (lib.nixvim)
+    defaultNullOpts
+    keymaps
+    mkPackageOption
+    mkCompositeOption
+    ;
+  types = lib.nixvim.nixvimTypes;
+
+  keymapsActions = {
     todoQuickFix = "TodoQuickFix";
     todoLocList = "TodoLocList";
     todoTrouble = "TodoTrouble";
     todoTelescope = "TodoTelescope";
   };
 in
-{
-  imports = [
-    (mkRemovedOptionModule [
-      "plugins"
-      "todo-comments"
-      "keymapsSilent"
-    ] "Use `plugins.todo-comments.keymaps.<COMMAND>.options.silent`.")
-  ];
-  options = {
-    plugins.todo-comments = helpers.neovim-plugin.extraOptionsOptions // {
-      enable = mkEnableOption "todo-comments";
+lib.nixvim.neovim-plugin.mkNeovimPlugin config {
+  name = "todo-comments";
+  originalName = "todo-comments.nvim";
+  defaultPackage = pkgs.vimPlugins.todo-comments-nvim;
 
-      package = helpers.mkPluginPackageOption "todo-comments" pkgs.vimPlugins.todo-comments-nvim;
+  maintainers = [ lib.maintainers.khaneliman ];
 
-      ripgrepPackage = helpers.mkPackageOption {
-        default = pkgs.ripgrep;
-        description = "Which package (if any) to be added for file search support in todo-comments.";
-      };
+  # TODO: Added 2024-08-16, remove after 24.11
+  imports =
+    let
+      basePluginPath = [
+        "plugins"
+        "todo-comments"
+      ];
+    in
+    [
+      (mkRemovedOptionModule (
+        basePluginPath ++ [ "keymapsSilent" ]
+      ) "Use `plugins.todo-comments.keymaps.<COMMAND>.options.silent`.")
+      (mkRenamedOptionModule (basePluginPath ++ [ "extraOptions" ]) (basePluginPath ++ [ "settings" ]))
```

Comment:
This is the same as `mkNeovimPlugin`'s `deprecateExtraOptions` option:
https://github.com/nix-community/nixvim/blob/78fc4be6a830e8dc01f3e66ddbe3243b4bfe8560/lib/neovim-plugin.nix#L25

I don't mind if you'd rather do it explicitly, but it might be easier to keep the dated `TODO` comments separate if we move this out of imports.

Reply:
The imports can be dated 2023-11-06 while `deprecateExtraOptions` and `optionsRenamedToSettings` can be grouped together and dated `2024-08-18`.